js/backup_in.js

```diff
@@ -332,16 +354,43 @@ function backup_generate()
             return;
         }
 
+        //electrum uses different xpriv version bytes for p2wpkh-p2sh BIP49  and p2wpkh BIP84
         var xpriv = '0488ade4' + '000000000000000000' + chaincode + '00' + privatekey;
         xpriv = base58check.encode(new Buffer(xpriv, 'hex'));
-       
+
         var bip32_keychain = new BIP32(xpriv);
-        var electrum_key = bip32_keychain.derive("m/44'/0'/0'");
-        electrum_key = base58check.encode(new Buffer(electrum_key.extended_private_key, 'hex'));
+        var electrum_key_leg = bip32_keychain.derive("m/44'/0'/0'");
+        electrum_key_leg = base58check.encode(new Buffer(electrum_key_leg.extended_private_key, 'hex'));
 
         var electrum_key_ms = bip32_keychain.derive("m/100'/45'/0'");
         electrum_key_ms = base58check.encode(new Buffer(electrum_key_ms.extended_private_key, 'hex'));
-        
+
+        var electrum_key_ltc_leg = bip32_keychain.derive("m/44'/2'/0'");
+        electrum_key_ltc_leg = base58check.encode(new Buffer(electrum_key_ltc_leg.extended_private_key, 'hex'));
+
+
+        var segwit_xpriv = '049d7878' + '000000000000000000' + chaincode + '00' + privatekey;
+        segwit_xpriv = base58check.encode(new Buffer(segwit_xpriv, 'hex'));
+
+        var bip32_keychain_seg = new BIP32(segwit_xpriv);
+        var electrum_key_seg = bip32_keychain_seg.derive("m/49'/0'/0'");
+        electrum_key_seg = base58check.encode(new Buffer(electrum_key_seg.extended_private_key, 'hex'));
+
+        var electrum_key_ltc_seg = bip32_keychain_seg.derive("m/49'/2'/0'");
+        electrum_key_ltc_seg = base58check.encode(new Buffer(electrum_key_ltc_seg.extended_private_key, 'hex'));
+
+
+        var native_xpriv = '04b2430c' + '000000000000000000' + chaincode + '00' + privatekey;
+        native_xpriv = base58check.encode(new Buffer(native_xpriv, 'hex'));
+
+        var bip32_keychain_nat_seg = new BIP32(native_xpriv);
+        var electrum_key_nat_seg = bip32_keychain_nat_seg.derive("m/84'/0'/0'");
+        electrum_key_nat_seg = base58check.encode(new Buffer(electrum_key_nat_seg.extended_private_key, 'hex'));
+
+        var electrum_key_ltc_nat_seg = bip32_keychain_nat_seg.derive("m/84'/2'/0'");
+        electrum_key_ltc_nat_seg = base58check.encode(new Buffer(electrum_key_ltc_nat_seg.extended_private_key, 'hex'));
```

Comment:
There's some code duplication and variable names that could be optimized (i.e. add "Btc" to variable names about BTC). Otherwise OK.

Reply:
I added the _btc_ , but removed it again for maximalist political reasons. 😛 
Can add it back, since btc is also named explicitly in js/bip32.js